packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -151,7 +154,8 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     commitData: CommitData,
     state: StreamState
   ): Promise<StreamState> {
-    // TODO: Assert that the 'prev' of the commit being applied is the end of the log in 'state'
+    StreamUtils.assertCommitLinksToState(state, commitData.commit)
```

Comment:
Wasn't there a bug (or still is) that the anchor service construction of the anchor commit doesn't include the `id` ?
Please investigate before merging.

Reply:
ooph yeah good call.  It's fixed in current CAS but did exist in the past so would affect some existing streams.  I'll have the check only apply if the `id` field actually exists